src/lapack/backends/cusolver/cusolver_scope_handle.cpp

```diff
@@ -39,12 +39,12 @@ thread_local cusolver_handle<pi_context> CusolverScopedContextHandler::handle_he
     cusolver_handle<pi_context>{};
 
 CusolverScopedContextHandler::CusolverScopedContextHandler(sycl::queue queue,
-                                                           sycl::interop_handler &ih)
+                                                           sycl::interop_handle &ih)
         : ih(ih),
           needToRecover_(false) {
-    placedContext_ = queue.get_context();
+    placedContext_ = new sycl::context(queue.get_context());
```

Comment:
Is there a reason placedContext was changed to a pointer? There is a very contrived scenario where a memory leak could occur. The cuda context get/set functions can throw and the user could recover from the exception, but placedContext_ will have leaked.

Reply:
Thanks for the question! This was done for performance reasons, following the same fix as in https://github.com/oneapi-src/oneMKL/pull/169. According to the description for the PR, it took a long time to create sycl::context with default constructor in cublas handler. I didn't run any performance tests, but I don't see a reason it would be different for the cuSOLVER backend. In #106 , you can see some performance comparisons for sgemm. The change significantly improved the performance of the oneMKL interfaces with cuBLAS backend.

Reply:
It does eliminate a call to the default constructor. I don't see how failing to get/set the cuda context would be recoverable anyways so I'll approve this as is.